helm-charts/common/retriever-usvc/templates/configmap.yaml

```diff
@@ -8,39 +8,45 @@ metadata:
 data:
-  {{- if .Values.MOSEC_EMBEDDING_ENDPOINT }}
-  MOSEC_EMBEDDING_ENDPOINT: {{ .Values.MOSEC_EMBEDDING_ENDPOINT | quote}}
-  MOSEC_EMBEDDING_MODEL: {{ .Values.MOSEC_EMBEDDING_MODEL | quote}}
-  {{- else if .Values.TEI_EMBEDDING_ENDPOINT }}
-  TEI_EMBEDDING_ENDPOINT: {{ .Values.TEI_EMBEDDING_ENDPOINT | quote }}
+  {{- if .Values.TEI_EMBEDDING_ENDPOINT }}
+  TEI_EMBEDDING_ENDPOINT: {{ tpl .Values.TEI_EMBEDDING_ENDPOINT . | quote }}
   {{- else if not .Values.LOCAL_EMBEDDING_MODEL }}
   TEI_EMBEDDING_ENDPOINT: "http://{{ .Release.Name }}-tei"
   {{- end }}
-  {{- if .Values.LOCAL_EMBEDDING_MODEL }}
-  EMBED_MODEL: {{ .Values.LOCAL_EMBEDDING_MODEL | quote }}
-  LOCAL_EMBEDDING_MODEL: {{ .Values.LOCAL_EMBEDDING_MODEL | quote }}
-  {{- end }}
-  {{- if .Values.REDIS_URL }}
-  REDIS_URL: {{ .Values.REDIS_URL | quote}}
+  {{- if eq "REDIS" .Values.RETRIEVER_BACKEND }}
+  RETRIEVER_COMPONENT_NAME: "OPEA_RETRIEVER_REDIS"
+  {{- if .Values.REDIS_HOST }}
+  REDIS_HOST: {{ tpl .Values.REDIS_HOST . | quote}}
   {{- else }}
-  REDIS_URL: "redis://{{ .Release.Name }}-redis-vector-db:6379"
+  REDIS_HOST: "{{ .Release.Name }}-redis-vector-db"
   {{- end }}
+  REDIS_PORT: {{ .Values.REDIS_PORT | quote }}
+  REDIS_SSL: {{ .Values.REDIS_SSL | quote }}
+  REDIS_PASSWORD: {{ .Values.REDIS_PASSWORD | quote }}
+  REDIS_USERNAME: {{ .Values.REDIS_USERNAME | quote }}
   INDEX_NAME: {{ .Values.INDEX_NAME | quote }}
+  EMBED_MODEL: {{ .Values.LOCAL_EMBEDDING_MODEL | quote }}
+  {{- else if eq "MILVUS" .Values.RETRIEVER_BACKEND }}
+  RETRIEVER_COMPONENT_NAME: "OPEA_RETRIEVER_MILVUS"
   {{- if .Values.MILVUS_HOST }}
-  MILVUS_HOST: {{ .Values.MILVUS_HOST | quote }}
+  MILVUS_HOST: {{ tpl .Values.MILVUS_HOST . | quote }}
   {{- else }}
   MILVUS_HOST: "{{ .Release.Name }}-milvus"
   {{- end }}
-  MILVUS: {{ .Values.MILVUS_HOST | quote }}
   MILVUS_PORT: {{ .Values.MILVUS_PORT | quote }}
-  {{- if .Values.COLLECTION_NAME }}
   COLLECTION_NAME: {{ .Values.COLLECTION_NAME | quote }}
+  LOCAL_EMBEDDING_MODEL: {{ .Values.LOCAL_EMBEDDING_MODEL | quote }}
+  {{- else }}
+  {{- cat "Invalid RETRIEVER_BACKEND:" .Values.RETRIEVER_BACKEND | fail }}
+  {{- end }}
+  {{- if .Values.global.HF_ENDPOINT }}
+  HF_ENDPOINT: {{ .Values.global.HF_ENDPOINT | quote}}
   {{- end }}
   EASYOCR_MODULE_PATH: "/tmp/.EasyOCR"
   http_proxy: {{ .Values.global.http_proxy | quote }}
   https_proxy: {{ .Values.global.https_proxy | quote }}
-  {{- if and (not .Values.REDIS_URL) (and (not .Values.TEI_EMBEDDING_ENDPOINT) (or .Values.global.http_proxy .Values.global.https_proxy)) }}
-  no_proxy: "{{ .Release.Name }}-tei,{{ .Release.Name }}-redis-vector-db,{{ .Values.global.no_proxy }}"
+  {{- if and (not .Values.REDIS_HOST) (and (not .Values.TEI_EMBEDDING_ENDPOINT) (or .Values.global.http_proxy .Values.global.https_proxy)) }}
+  no_proxy: "{{ .Release.Name }}-tei,{{ .Release.Name }}-redis-vector-db,{{ .Release.Name }}-milvus,{{ .Values.global.no_proxy }}"
```

Comment:
This adds Milvus host to `no_proxy`, so shouldn't the if check also for MILVUS_PORT?

----
Btw. I would probably do myself just something like this:
```
{{- if .Values.global.no_proxy }}
// use user provided proxy
{{- else }}
// use above proxy setting
{{- end }}
```

(Having those items in `no_proxy` is unlikely to cause any harm.  Vector DBs are supposed to store /retrieve data, not query internet themselves...)

Reply:
This is not for vector DB itself, but for dataprep and/or retriever to talk to vectorDB. The reason behind this logic is that if the end user doesn't care about where is the vectorDB(e.g. bring up the vector DB automatically as the default config within the same k8s cluster), we should help them automatically populate the DB HOST info into no_proxy. Otherwise, if the user deploy the vector DB somewhere else, they need to set vectorDB HOST by themself and hence the no_proxy explicity.